lib/shoryuken/fetcher.rb

```diff
@@ -44,6 +48,36 @@ def fetch_with_auto_retry(max_attempts)
       end
     end
 
+    def regular_fetch(queue, limit)
+      sqs_msgs = Array(receive_messages(queue, limit))
+      logger.debug { "Found #{sqs_msgs.size} messages for #{queue.name}" } unless sqs_msgs.empty?
+      sqs_msgs
+    end
+
+    def fetch_with_batch_options(queue, limit, batch_max_size, batch_timeout)
+      batch = Shoryuken::MessageBatch.new(max_size: batch_max_size, timeout: batch_timeout)
+      loop do
+        check_visibility_timeout(queue, batch.timeout)
+        sqs_msgs = Array(receive_messages(queue, [limit, batch.max_size, batch.max_size - batch.size].min))
+        logger.debug { "Found #{sqs_msgs.size} messages for #{queue.name}" } unless sqs_msgs.empty?
+        sqs_msgs.each { |sqs_msg| batch.add_message!(sqs_msg) }
+        break if batch.full? || batch.timeout_expired?
+      end
+      batch.messages
+    end
+
+    def check_visibility_timeout(queue, batch_timeout)
+      return if @visibility_timeout_ok
```

Comment:
nit: If `@visibility_timeout_ok` is being used for memorization, we usually suffix with `_`. 

```suggestion
      return if @_visibility_timeout_ok
```